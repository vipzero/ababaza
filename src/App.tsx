import React from 'react'
import {
	Container,
	CssBaseline,
	Typography,
	Button,
	createMuiTheme,
} from '@material-ui/core'
import { ThemeProvider } from '@material-ui/styles'
import styled from 'styled-components'
import abaImg from './ababazaminal.png'
import aImg from './a.png'

const Title = styled(Typography)`
	font-weight: 800 !important;
	--font-sans: ;
`

const theme = createMuiTheme({
	typography: {
		fontFamily: `-apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
		'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
		sans-serif`,
	},
	overrides: {
		MuiButton: {
			root: {
				textTransform: 'none',
				color: 'gray',
				padding: '8px 56px !important',
			},
			outlined: {
				background: 'black',
				color: 'white',
				'&:hover': {
					background: 'black',
				},
			},
		},
	},
})

const Sign = styled.div`
	background: black;
	color: white;
	width: 40px;
	height: 40px;
	border-radius: 8px;
	text-align: center;
	padding: 4px;
`

const Item = ({ sign, char }: { sign: string; char: string }) => (
	<div style={{ width: '260px', margin: '30px' }}>
		<Sign>
			<Typography variant="h5">{sign}</Typography>
		</Sign>
		<Typography variant="h5" style={{ fontWeight: 'bold', marginTop: '10px' }}>
			{char} of Ababaza
		</Typography>
		<Typography variant="body1" style={{ color: '#444' }}>
			Aba baza Aba baza
		</Typography>
	</div>
)

const EyeCatch = styled.div`
	margin-top: 40px;
	width: 100%;
	display: flex;
	justify-content: center;
	align-items: center;
	div {
		width: 460px;
	}
	${theme.breakpoints.down('sm')} {
		flex-direction: column;
		text-align: center;
		align-content: center;
		div {
			width: 100%;
		}
	}
`

const Header = styled.header`
	padding: 24px;
	border-bottom: 1px solid #ccc;
	img {
		width: 30px;
	}
	p {
		padding-top: 4px;
		color: #444;
		padding-left: 16px;
	}
`

const Row = styled.div`
	display: flex;
	justify-content: space-evenly;
	${theme.breakpoints.down('sm')} {
		flex-direction: column;
		align-content: center;
	}
`

const ScreenShot = styled.img`
	width: 100%;
	${theme.breakpoints.down('sm')} {
	}
`

function App() {
	return (
		<ThemeProvider theme={theme}>
			<CssBaseline />
			<Header>
				<div style={{ display: 'flex' }}>
					<img src={aImg} />
					<Typography>Ababaza</Typography>
					<Typography>Bazaza</Typography>
					<Typography>Aza</Typography>
				</div>
			</Header>
			<Container>
				<EyeCatch>
					<div style={{ paddingTop: '20px' }}>
						<Title variant="h2" component="h1">
							<span style={{ color: '#444' }}>Ababaza</span>
							<br />A Bab Aza
						</Title>
						<Typography
							variant="h6"
							style={{ color: '#555', margin: '20px 0' }}
						>
							Ababaza babaza baza za {'&'} abaza aza bazaba.
						</Typography>
						<div style={{ display: 'flex', justifyContent: 'center' }}>
							<Button size="large" variant="outlined">
								Aba baza →
							</Button>
							<Button size="large">Ababa zababaza.</Button>
						</div>
					</div>
					<div>
						<ScreenShot src={abaImg}></ScreenShot>
					</div>
				</EyeCatch>
				<div style={{ padding: '40px 0' }}>
					<Row>
						<Item sign="∋" char="A" />
						<Item sign="∀" char="B" />
					</Row>
					<Row>
						<Item sign="∠" char="B'" />
						<Item sign="×" char="Z" />
					</Row>
				</div>
			</Container>
		</ThemeProvider>
	)
}

export default App
