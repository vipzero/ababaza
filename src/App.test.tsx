import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import App from './App'

test('renders without crashing', () => {
	const div = document.createElement('div')

	ReactDOM.render(<App />, div)
	ReactDOM.unmountComponentAtNode(div)

	const tree = renderer.create(<App />).toJSON()

	expect(tree).toMatchSnapshot()
})
