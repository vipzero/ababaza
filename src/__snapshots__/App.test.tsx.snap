// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  .c0 {
  padding: 24px;
  border-bottom: 1px solid #ccc;
}

.c0 img {
  width: 30px;
}

.c0 p {
  padding-top: 4px;
  color: #444;
  padding-left: 16px;
}

@media (max-width:959.95px) {

}

@media (max-width:959.95px) {

}

@media (max-width:959.95px) {

}

<header
    className="c0"
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <img
        src="a.png"
      />
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Ababaza
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Bazaza
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Aza
      </p>
    </div>
  </header>,
  .c1 {
  font-weight: 800 !important;
}

.c4 {
  background: black;
  color: white;
  width: 40px;
  height: 40px;
  border-radius: 8px;
  text-align: center;
  padding: 4px;
}

.c0 {
  margin-top: 40px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 div {
  width: 460px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c2 {
  width: 100%;
}

@media (max-width:959.95px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: center;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }

  .c0 div {
    width: 100%;
  }
}

@media (max-width:959.95px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

@media (max-width:959.95px) {

}

<div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="c0"
    >
      <div
        style={
          Object {
            "paddingTop": "20px",
          }
        }
      >
        <h1
          className="MuiTypography-root c1 MuiTypography-h2"
        >
          <span
            style={
              Object {
                "color": "#444",
              }
            }
          >
            Ababaza
          </span>
          <br />
          A Bab Aza
        </h1>
        <h6
          className="MuiTypography-root MuiTypography-h6"
          style={
            Object {
              "color": "#555",
              "margin": "20px 0",
            }
          }
        >
          Ababaza babaza baza za 
          &
           abaza aza bazaba.
        </h6>
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Aba baza →
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Ababa zababaza.
            </span>
          </button>
        </div>
      </div>
      <div>
        <img
          className="c2"
          src="ababazaminal.png"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "40px 0",
        }
      }
    >
      <div
        className="c3"
      >
        <div
          style={
            Object {
              "margin": "30px",
              "width": "260px",
            }
          }
        >
          <div
            className="c4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              ∋
            </h5>
          </div>
          <h5
            className="MuiTypography-root MuiTypography-h5"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": "10px",
              }
            }
          >
            A
             of Ababaza
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "color": "#444",
              }
            }
          >
            Aba baza Aba baza
          </p>
        </div>
        <div
          style={
            Object {
              "margin": "30px",
              "width": "260px",
            }
          }
        >
          <div
            className="c4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              ∀
            </h5>
          </div>
          <h5
            className="MuiTypography-root MuiTypography-h5"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": "10px",
              }
            }
          >
            B
             of Ababaza
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "color": "#444",
              }
            }
          >
            Aba baza Aba baza
          </p>
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          style={
            Object {
              "margin": "30px",
              "width": "260px",
            }
          }
        >
          <div
            className="c4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              ∠
            </h5>
          </div>
          <h5
            className="MuiTypography-root MuiTypography-h5"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": "10px",
              }
            }
          >
            B'
             of Ababaza
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "color": "#444",
              }
            }
          >
            Aba baza Aba baza
          </p>
        </div>
        <div
          style={
            Object {
              "margin": "30px",
              "width": "260px",
            }
          }
        >
          <div
            className="c4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              ×
            </h5>
          </div>
          <h5
            className="MuiTypography-root MuiTypography-h5"
            style={
              Object {
                "fontWeight": "bold",
                "marginTop": "10px",
              }
            }
          >
            Z
             of Ababaza
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "color": "#444",
              }
            }
          >
            Aba baza Aba baza
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
